CRM/Utils/Check.php

```diff
@@ -209,7 +212,7 @@ public static function checkAll($max = FALSE) {
 
     foreach ($messages as $key => $message) {
       $hush = self::checkHushSnooze($message);
-      $message->setVisible(!$hush);
+      $messages[$key]->setVisible(!$hush);
```

Comment:
Wonder if it wouldn't be better to revert this line back, but add an ampersand in front of $messages in the foreach loop to explicitly pass it by reference. That ought to prevent the loop from confusingly making copies of the objects while iterating.